extensions/outline-document-search/src/browse-collections.tsx

```diff
@@ -0,0 +1,171 @@
+  const [selectedCollection, setSelectedCollection] = useState<Collection | null>(null);
+  const [documents, setDocuments] = useState<OutlineDocument[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    if (!instances || instances.length === 0) {
+      setIsLoading(false);
+      return;
+    }
+
+    // If only one instance, select it automatically
+    if (instances.length === 1) {
+      handleSelectInstance(instances[0]);
+    } else {
+      setIsLoading(false);
+    }
+  }, [instances]);
+
+  async function handleSelectInstance(instance: Instance) {
+    setSelectedInstance(instance);
+    setIsLoading(true);
+    try {
+      const api = new OutlineApi(instance);
+      const cols = await api.listCollections();
+      setCollections(cols);
+    } catch (error) {
+      await showToast({
+        style: Toast.Style.Failure,
+        title: "Failed to load collections",
+        message: error instanceof Error ? error.message : String(error),
+      });
+    } finally {
+      setIsLoading(false);
+    }
+  }
+
+  async function handleSelectCollection(collection: Collection) {
+    if (!selectedInstance) return;
+
+    setSelectedCollection(collection);
+    setIsLoading(true);
+    try {
+      const api = new OutlineApi(selectedInstance);
+      const docs = await api.getCollectionDocuments(collection.id);
+      setDocuments(docs);
+    } catch (error) {
+      await showToast({
+        style: Toast.Style.Failure,
+        title: "Failed to load documents",
+        message: error instanceof Error ? error.message : String(error),
+      });
+    } finally {
+      setIsLoading(false);
+    }
+  }
+
+  function handleBack() {
+    if (selectedCollection) {
+      setSelectedCollection(null);
+      setDocuments([]);
+    } else if (selectedInstance && instances && instances.length > 1) {
+      setSelectedInstance(null);
+      setCollections([]);
+    }
+  }
+
+  if (!instances || instances.length === 0) {
+    return (
+      <List>
+        <EmptyList />
+      </List>
+    );
+  }
+
+  // Show documents in selected collection
+  if (selectedCollection && selectedInstance) {
+    return (
+      <List
+        isLoading={isLoading}
+        searchBarPlaceholder={`Search in ${selectedCollection.name}...`}
+        navigationTitle={selectedCollection.name}
+      >
+        <List.Item
+          title="← Back to Collections"
+          icon={Icon.ArrowLeft}
+          actions={
+            <ActionPanel>
+              <Action title="Back" onAction={handleBack} />
+            </ActionPanel>
+          }
+        />
+        {documents.map((doc) => (
+          <List.Item
+            key={doc.id}
+            title={doc.title}
+            subtitle={(doc.text || "").substring(0, 50) + (doc.text && doc.text.length > 50 ? "..." : "")}
+            icon={Icon.Document}
+            actions={<DocumentActions document={doc} instance={selectedInstance} />}
+          />
```

Comment:
Can we have `ViewDocument.tsx` here, instead? This way, we would have a document preview similar to the document search.